src/components/common/EmailPrompt.vue

```diff
@@ -0,0 +1,135 @@
+            required: false,
+            default: "Add User Email"
+        },
+        dialog: {
+            type: String,
+            required: false,
+            default: "Your email is required to use this feature. Please add it below"
+        },
+        isFirstLoginPrompt: {
+            type: Boolean,
+            required: true
+        }
+    },
+
+    setup: () => {
+        const { item: user, setItemId: setUserId, saveItem: saveUser } = useItem({ itemTypeName: 'me' })
+        const visible = ref(false);
+
+        return {
+            user,
+            setUserId,
+            saveUser,
+            visible,
+        }
+    },
+
+    data: function() {
+        return {
+            email: null,
+            emailValidationError: null
+        }
+    },
+
+    watch: {
+        user: {
+            handler: function() {
+                this.email = this.user?.email
+
+                if (this.$props.isFirstLoginPrompt) {
+                    this.visible = this.user.isFirstLogin && this.user.email == null
+                }
+                else {
+                    this.visible = this.user.email == null
+                }
+            }
+        }
+    },
+
+    mounted: function() {
+        this.setUserId('me')
+    },
+
+    methods: {
+        saveEmail: async function() {
+            const email = this.email ? this.email.trim() : null
+
+            if (!email) {
+                this.emailValidationError = "Email cannot be empty."
+            }
+
+            if (this.user && email && email != this.user.email) {
+                let errorResponse = null
+                try {
+                    await this.saveUser({
+                        item: {
+                            ...this.user,
+                            email
+                        }
+                    })
+                } catch (error) {
+                    errorResponse = error.response
+                }
+
+                if (errorResponse) {
+                    if (typeof errorResponse.data.detail === 'string' || errorResponse.data.detail instanceof String) {
+                        // Handle generic errors that are not surfaced by the API as objects
+                        this.$toast.add({ severity: 'error', summary: `Encountered an error saving email: ${errorResponse.data.detail}` })
+                    }
+                    else {
+                        // There will only be one validation error surfaced for emails.
+                        this.emailValidationError = errorResponse.data.detail[0].msg
+                    }
+                }
+                else {
+                    this.emailValidationError = null
+                    this.user.email = email
+                    this.$toast.add({ severity: 'success', summary: `Your email was succesfully updated.`, life: 3000 })
+                    this.visible = false
+                }
+            }
+        },
+
+        ignoreEmail: async function() {
```

Comment:
Yeah, maybe this should get reworked a little tbh. It saves the user so that their `is_first_login` value in the backend gets set to `False`. That ensures that the component will only show up once for their first login. 

It's kinda relying on a side-effect of the update user endpoint though... (and we probably also would prefer to have the email field in the endpoint schema to not be optional). Maybe a dedicated endpoint would be better rather than this?

Reply:
yeah I would have an api side mechanism to set the new column to true directly